cpp/tests/utilities/column_wrapper.hpp

```diff
@@ -423,13 +423,55 @@ class fixed_width_column_wrapper : public detail::column_wrapper {
     : fixed_width_column_wrapper(begin, end, std::cbegin(validity))
   {
   }
+
+  /**
+   * @brief Construct a nullable column from a list of pairs of fixed-width
+   * elements and validity booleans of each element.
+   *
+   * The validity of each element is determined by the boolean element in the pair
+   * where `true` indicates the element is valid, and `false` indicates the
+   * element is null.
+   *
+   * Example:
+   * @code{.cpp}
+   * // Creates a nullable INT32 column with 4 elements: {1, NULL, 3, NULL}
+   * using p = std::pair<int32_t, bool>;
+   * fixed_width_column_wrapper<int32_t> w( p{1, true}, p{2, false}, p{3, true}, p{4, false} );
+   * @endcode
+   *
+   * @param elements The list of pairs of element and validity booleans
+   */
+  template <typename ElementFrom>
+  fixed_width_column_wrapper(std::initializer_list<std::pair<ElementFrom, bool>> elements)
```

Comment:
Remind me why we didn't go with the `pair_builder` approach to avoid needing the explicit `p` type?

Reply:
The `pair_builder` approach ran into issues with the compiler being unable to determine how to handle empty initializer lists. Things like `cudf::test::fixed_width_column_wrapper<bool>({});` would result in ambiguity. Worse, it was picking up things like `auto column = cudf::test::fixed_width_column_wrapper<int>{{4, 0}}` and attempting to feed it into the `pair_builder`. I really wanted that one to work because I think it looks much nicer to the user, but it ran into issue after issue and I was concerned that there would be a silently failing issues that snuck through.

Reply:
Understood. As you've learned, `initializer_list` constructors can prove to be quite a headache to get right, especially if you have more than one. This solution seems like the best compromise.